src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -357,66 +354,64 @@ namespace
                     abort_switch,
                     batch_size,
                     framebuffer,
-                    second_framebuffer,
+                    second_framebuffer.get(),
                     tile_origin_x,
                     tile_origin_y,
                     frame,
                     frame_width,
                     pass_hash,
                     aov_count);
 
-                // Check if this was the last batch.
-                if (remaining_samples - batch_size < 1)
+                const AABB2u& block_image_bb = AABB2i::intersect(framebuffer->get_crop_window(), pb.m_surface);
```

Comment:
`AABB2i::intersect()` returns a bounding box by copy. I find it best to acknowledge that fact by not keeping a const reference to the result:

    const AABB2u block_image_bb = AABB2i::intersect(framebuffer->get_crop_window(), pb.m_surface);

Reply:
To be clear:

Your code was valid because C++ extends the lifetime of temporaries to match the lifetime of const references to them:

https://blog.galowicz.de/2016/03/23/const_reference_to_temporary_object/

Reply:
good to know, thanks